File: Itinere_3/Laravel/department_employee/app/Http/Controllers/EmployeeController.php
<?php

namespace App\Http\Controllers;

use App\Models\Employee;
use App\Models\Department;
use Illuminate\Http\Request;
use SebastianBergmann\CodeCoverage\Util\Percentage;

class EmployeeController extends Controller
{
    /**
     * Display a listing of the resource.
     */
    public function index()
    {
        return view("employees.index", ["employees" => Employee::all()]);
    }

    /**
     * Show the form for creating a new resource.
     */
    public function create()
    {
        return view("employees.create", ["departments" => Department::all()]);
    }

    /**
     * Store a newly created resource in storage.
     */
    public function store(Request $request)
    {
        Employee::create($request->all());
        return redirect()->route("employees.index");
    }

    /**
     * Display the specified resource.
     */
    public function show(Employee $employee)
    {
        return view("employees.show", ["e" => $employee]);
    }

    /**
     * Show the form for editing the specified resource.
     */
    public function edit(Employee $employee)
    {
        return view("employees.edit", ["e" => $employee, "departments" => Department::all()]);
    }

    /**
     * Update the specified resource in storage.
     */
    public function update(Request $request, Employee $employee)
    {
        $employee->update($request->all());
        return redirect()->route("employees.index");
    }

    /**
     * Remove the specified resource from storage.
     */
    public function destroy(Employee $employee)
    {
        $employee->delete();
        return redirect()->route("employees.index");
    }

    public function increaseSalary(Employee $employee, $percentage)
    {
        $employee->salary = $employee->salary * (1 + $percentage/100);
        $employee->save();
    
        return redirect()->route('employees.show', $employee);
    }
}

File: Itinere_3/Spring-Boot/department_employee/src/main/java/com/example/department_employee/controllers/EmployeeController.java
package com.example.department_employee.controllers;

import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;

import com.example.department_employee.Repositories.EmployeeRepository;
import com.example.department_employee.Repositories.DepartmentReposity;
import com.example.department_employee.models.Employee;

@Controller
public class EmployeeController {
    
    private final EmployeeRepository repo;
    private final DepartmentReposity departmentRepo;


    public EmployeeController(EmployeeRepository repo, DepartmentReposity departmentRepo) {
        this.repo = repo;
        this.departmentRepo = departmentRepo;
    }

    @GetMapping("/employees")
    String index(Model model){
        model.addAttribute("employees", repo.findAll());
        return "/employees/index";
    }

    @GetMapping("/employees/create")
    String create(Model model){
        model.addAttribute("departments", departmentRepo.findAll());
        return "/employees/create";
    }

    @PostMapping("/employees/store")
    String store(Employee employee){
        repo.save(employee);
        return "redirect:/employees";
    }

    @GetMapping("/employees/show")
    String show(Model model, Long id){
        model.addAttribute("e", repo.findById(id).orElse(null));
        return "/employees/show";
    }

    @GetMapping("/employees/edit")
    String edit(Model model, Long id){
        model.addAttribute("departments", departmentRepo.findAll());
        model.addAttribute("e", repo.findById(id).orElse(null));
        return "/employees/edit";
    }

    @PostMapping("/employees/update")
    String update(Employee employee){
        repo.save(employee);
        return "redirect:/employees";
    }

    @PostMapping("/employees/destroy")
    String destroy(Employee employee){
        repo.deleteById(employee.getId());
        return "redirect:/employees";
    }

    @GetMapping("/employees/increaseSalary")
    String increaseSalary(Integer percentage, Model model, Long id){
        repo.increaseSalary(id, percentage);
        return show(model, id);
    }
}   
